tests/unit/anchore_engine/services/policy_engine/engine/feeds/test_grypedb_sync.py

```diff
@@ -50,6 +52,34 @@ def _mock(mock_active_dbs=[], mock_local_checksum=""):
 
         return _mock
 
+    @pytest.fixture
+    def setup_for_thread_testing(self, mock_calls_for_sync):
+        """
+        2 tests that run tests for threads and locking use the same mock data captured in this fixture
+        """
+        old_checksum = (
+            "eef3b1bcd5728346cb1b30eae09647348bacfbde3ba225d70cb0374da249277c"
+        )
+        new_checksum = (
+            "366ab0a94f4ed9c22f5cc93e4d8f6724163a357ae5190740c1b5f251fd706cc4"
+        )
+
+        # mock initial state so execution occurs
+        mock_calls_for_sync([GrypeDBMetadata(checksum=old_checksum)], "")
+
+        # Mock the update_grypedb method for task to sleep and update mocks for active and local grype dbs
+        def _mock_update_grypedb_for_thread1(
+            active_grypedb=None, grypedb_file_path=None
+        ):
+            # sleep to ensure lock is taken
+            time.sleep(10)
+
+            # mock the returns to mimic persistent change of active grypedb local and global
+            # This in effect mocks the actual execution for the first thread
+            mock_calls_for_sync([GrypeDBMetadata(checksum=new_checksum)], new_checksum)
+
+        GrypeDBSyncManager._update_grypedb = _mock_update_grypedb_for_thread1
```

Comment:
This is called _"patching"_ and it is effectively overriding what `._update_grypedb` is, but it isn't setting it back like it was before. This has consequences in failures or when other tests depend on `._update_grypedb` being set in a different way. 

I recommend using `monkeypatch` which is a Pytest fixture or if you really must, you can do this with `mock` as well. The [Monkeypatching chapter](https://paiml.com/docs/home/books/testing-in-python/chapter08-monkeypatching/) in the Testing In Python book might be helpful if you are unsure

Reply:
This is super useful! I actually was running into this issue and had not realized this stye of patching persisted outside of the context it was defined! Using the monkeypatch fixed it!